test/unit/io/stream/fast_streambuf_iterator_test.cpp

```diff
@@ -0,0 +1,204 @@
+    EXPECT_FALSE(std::bidirectional_iterator<type>);
+    EXPECT_FALSE(std::random_access_iterator<type>);
+}
+
+TEST(fast_ostreambuf_iterator, construction)
+{
+    using type = seqan3::detail::fast_ostreambuf_iterator<char>;
+
+    EXPECT_TRUE(std::is_nothrow_default_constructible_v<type>);
+    EXPECT_TRUE(std::is_nothrow_copy_constructible_v<type>);
+    EXPECT_TRUE(std::is_nothrow_move_constructible_v<type>);
+    EXPECT_TRUE(std::is_nothrow_copy_assignable_v<type>);
+    EXPECT_TRUE(std::is_nothrow_move_assignable_v<type>);
+    EXPECT_TRUE(std::is_nothrow_destructible_v<type>);
+    EXPECT_TRUE((std::is_constructible_v<type, std::basic_streambuf<char> &>));
+}
+
+TEST(fast_ostreambuf_iterator, assignment)
+{
+    std::ostringstream ostr{};
+    seqan3::detail::fast_ostreambuf_iterator<char> it{*ostr.rdbuf()};
+
+    it = 't';
+    *it = 'e';
+
+    it++; // no op
+    ++it; // no op
+
+    *it++ = 's';
+    *++it = 't';
+
+    EXPECT_EQ(ostr.str(), std::string_view{"test"});
+}
+
+TEST(fast_ostreambuf_iterator, write_range)
+{
+    { // the simple case without using the return value of write_value
+        std::ostringstream ostr{};
+        seqan3::detail::fast_ostreambuf_iterator<char> it{*ostr.rdbuf()};
+
+        std::string rng{"test\ntestest"};
+
+        it.write_range(rng);
+
+        EXPECT_EQ(ostr.str(), std::string_view{"test\ntestest"});
+    }
+
+    { // ensure that buffer overflows at least once
+        std::ostringstream ostr;
+        char buf[40];
+        ostr.rdbuf()->pubsetbuf(buf, sizeof buf);
+
+        seqan3::detail::fast_ostreambuf_iterator<char> it{*ostr.rdbuf()};
+
+        std::string rng{"veryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryverylargerange"};
+
+        it.write_range(rng);
+
+        EXPECT_EQ(ostr.str().substr(0, 78),
+                  std::string_view{"veryveryveryveryveryveryveryveryveryveryveryveryveryveryveryveryverylargerange"});
+    }
+
+    { // use the return value of write_range to keep track of the chunk you have already written
+        std::ostringstream ostr{};
+        seqan3::detail::fast_ostreambuf_iterator<char> it{*ostr.rdbuf()};
+
+        std::string rng{"test\ntestest"};
+        using str_it_t = std::string::iterator;
+
+        auto rng_it = it.write_range(std::ranges::subrange<str_it_t, str_it_t>(rng.begin(), rng.begin() + 5));
+
+        EXPECT_EQ(rng_it, (rng.begin() + 5));
+        EXPECT_EQ(ostr.str(), std::string_view{"test\n"});
+
+        rng_it = it.write_range(std::ranges::subrange<str_it_t, str_it_t>(rng.begin() + 5, rng.end()));
+
+        EXPECT_EQ(rng_it, rng.end());
+        EXPECT_EQ(ostr.str(), std::string_view{"test\ntestest"});
+    }
+
+    { // write_range on a non-forwarding range returns void
+        std::ostringstream ostr{};
+        seqan3::detail::fast_ostreambuf_iterator<char> it{*ostr.rdbuf()};
+
+        EXPECT_TRUE((std::same_as<void, decltype(it.write_range(std::string{"foo"}))>));
```

Comment:
Please write tests that executes something and see what happens, not just syntax checking.

My guess is an infinite loop :)

Reply:
There is a test case here:
https://github.com/seqan/seqan3/pull/1472/files#diff-39898526460e7bc8f04d2d99e96e67feR109

Reply:
no infinite loop

Reply:
I added an test. no infinite loop

Reply:
Where?

Reply:
wait 'til I push